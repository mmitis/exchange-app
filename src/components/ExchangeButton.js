//@flow
import React from "react";

type Props = {
    label : string,
    isDisabled: boolean,
    onClick: () => void
}
export const ExchangeButton = ({
                                  onClick,
                                  isDisabled,
                                  label
} : Props) => {

    const classNames = !isDisabled ? 'exchange-button' : 'exchange-button exchange-button--disabled';
    return (
     <button 
        onClick={() => onClick()}
        className={classNames}
        disabled={isDisabled} >
            {label}
    </button>
    )
};

export default React.memo(ExchangeButton);
