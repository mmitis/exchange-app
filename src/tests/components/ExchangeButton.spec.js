import ExchangeButton from './../../components/ExchangeButton';
import { mount } from 'enzyme';
import React from "react";

describe('Component: ExchangeButton', () => {

    it('test simple render', async () => {
        const button = mount(<ExchangeButton
            onClick={() => {}}
            label={'Exchange!'}
            isDisabled={true}
        />);

        expect(button).toMatchSnapshot();

    });

    it('test simple render - disabled', async () => {
              const button = mount(<ExchangeButton
            onClick={() => {}}
            label={'Exchange!'}
            isDisabled={false}
        />);

        expect(button).toMatchSnapshot();

    });
});
