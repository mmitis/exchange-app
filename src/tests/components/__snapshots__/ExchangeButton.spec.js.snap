// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ExchangeButton test simple render - disabled 1`] = `ReactWrapper {}`;

exports[`Component: ExchangeButton test simple render 1`] = `ReactWrapper {}`;
